common/races.h

```diff
@@ -885,7 +772,534 @@ namespace Race {
+	constexpr uint16 Ogre      = 10;
+	constexpr uint16 Halfling  = 11;
+	constexpr uint16 Gnome     = 12;
+	constexpr uint16 Iksar     = 13;
+	constexpr uint16 VahShir   = 14;
+	constexpr uint16 Froglok   = 15;
+	constexpr uint16 Drakkin   = 16;
+	constexpr uint16 Count     = 16;
+}
+
+namespace PlayerRaceBitmask {
+	constexpr uint16 Unknown   = 0;
+	constexpr uint16 Human     = 1;
+	constexpr uint16 Barbarian = 2;
+	constexpr uint16 Erudite   = 4;
+	constexpr uint16 WoodElf   = 8;
+	constexpr uint16 HighElf   = 16;
+	constexpr uint16 DarkElf   = 32;
+	constexpr uint16 HalfElf   = 64;
+	constexpr uint16 Dwarf     = 128;
+	constexpr uint16 Troll     = 256;
+	constexpr uint16 Ogre      = 512;
+	constexpr uint16 Halfling  = 1024;
+	constexpr uint16 Gnome     = 2048;
+	constexpr uint16 Iksar     = 4096;
+	constexpr uint16 VahShir   = 8192;
+	constexpr uint16 Froglok   = 16384;
+	constexpr uint16 Drakkin   = 32768;
+	constexpr uint16 All       = 65535;
+}
+
+static std::map<uint16, uint16> player_race_bitmasks = {
+	{ Race::Human,     PlayerRaceBitmask::Human },
+	{ Race::Barbarian, PlayerRaceBitmask::Barbarian },
+	{ Race::Erudite,   PlayerRaceBitmask::Erudite },
+	{ Race::WoodElf,   PlayerRaceBitmask::WoodElf },
+	{ Race::HighElf,   PlayerRaceBitmask::HighElf },
+	{ Race::DarkElf,   PlayerRaceBitmask::DarkElf },
+	{ Race::HalfElf,   PlayerRaceBitmask::HalfElf },
+	{ Race::Dwarf,     PlayerRaceBitmask::Dwarf },
+	{ Race::Troll,     PlayerRaceBitmask::Troll },
+	{ Race::Ogre,      PlayerRaceBitmask::Ogre },
+	{ Race::Halfling,  PlayerRaceBitmask::Halfling },
+	{ Race::Gnome,     PlayerRaceBitmask::Gnome },
+	{ Race::Iksar,     PlayerRaceBitmask::Iksar },
+	{ Race::VahShir,   PlayerRaceBitmask::VahShir },
+	{ Race::Froglok,   PlayerRaceBitmask::Froglok },
+	{ Race::Drakkin,   PlayerRaceBitmask::Drakkin },
+};
+
+static std::map<uint16, std::string> player_race_abbreviations = {
+	{ Race::Human,     "HUM" },
+	{ Race::Barbarian, "BAR" },
+	{ Race::Erudite,   "ERU" },
+	{ Race::WoodElf,   "ELF" },
+	{ Race::HighElf,   "HIE" },
+	{ Race::DarkElf,   "DEF" },
+	{ Race::HalfElf,   "HEF" },
+	{ Race::Dwarf,     "DWF" },
+	{ Race::Troll,     "TRL" },
+	{ Race::Ogre,      "OGR" },
+	{ Race::Halfling,  "HFL" },
+	{ Race::Gnome,     "GNM" },
+	{ Race::Iksar,     "IKS" },
+	{ Race::VahShir,   "VAH" },
+	{ Race::Froglok2,  "FRG" },
+	{ Race::Drakkin,   "DRK" }
+};
+
+static const std::map<uint16, std::vector<float>> race_sizes = {
+	{ Race::Human,                 { 7.0f,   7.0f }},
+	{ Race::Barbarian,             { 5.0f,   5.0f }},
+	{ Race::Erudite,               { 5.0f,   5.0f }},
+	{ Race::WoodElf,               { 5.5f,   5.5f }},
+	{ Race::HighElf,               { 4.0f,   4.0f }},
+	{ Race::DarkElf,               { 8.0f,   8.0f }},
+	{ Race::HalfElf,               { 9.0f,   9.0f }},
+	{ Race::Dwarf,                 { 3.5f,   3.5f }},
+	{ Race::Troll,                 { 3.0f,   3.0f }},
+	{ Race::Ogre,                  { 2.0f,   2.0f }},
```

Comment:
Can we show our work to show that this list is 100% accurate? This can create all kinds of headache if we get this wrong

Reply:
I wrote a script to compare old / new. But without even using the script I see that the first set of entries is wrong

```
	static float male_height[] = {
	    6.0f,  6.0f,  7.0f,   6.0f,   5.0f,  6.0f,  5.0f,   5.5f,  4.0f,  8.0f,  9.0f,  3.5f,  3.0f,  6.0f,   6.0f,
```

```
	static float female_height[] = {
	    6.0f,  6.0f,  7.0f,   6.0f,   5.0f,  6.0f,  5.0f,   5.5f,  4.0f,  8.0f,  9.0f,  3.5f,  3.0f,  6.0f,   6.0f,
```

In the new array we are 7,5,5

Not a good sign

Reply:
My script I used did not put a unique case for `6.0` since it is the default male and female size for races, so any race that had `6.0` is not specified in the new data and instead uses the default.

Reply:
I can post the code and explain when at home.